src/interfaces/IIdRegistry.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+interface IIdRegistry {
+    /*//////////////////////////////////////////////////////////////
+                              CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Contract version specified in the Farcaster protocol version scheme.
+     */
+    function VERSION() external view returns (string memory);
+
+    /**
+     * @notice EIP-712 typehash for Register signatures.
+     */
+    function REGISTER_TYPEHASH() external view returns (bytes32);
+
+    /**
+     * @notice EIP-712 typehash for Transfer signatures.
+     */
+    function TRANSFER_TYPEHASH() external view returns (bytes32);
+
+    /**
+     * @notice EIP-712 typehash for ChangeRecoveryAddress signatures.
+     */
+    function CHANGE_RECOVERY_ADDRESS_TYPEHASH() external view returns (bytes32);
+
+    /*//////////////////////////////////////////////////////////////
+                                 STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice The last Farcaster id that was issued.
+     */
+    function idCounter() external view returns (uint256);
+
+    /**
+     * @notice Maps each address to an fid, or zero if it does not own an fid.
+     */
+    function idOf(address owner) external view returns (uint256 fid);
+
+    /**
+     * @notice Maps each fid to an address that can initiate a recovery.
+     */
+    function recoveryOf(uint256 fid) external view returns (address recovery);
+
+    /*//////////////////////////////////////////////////////////////
+                             REGISTRATION LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /**
```

Comment:
is the desired outcome to duplicate comments across the interface and implementation?

this seems like it has the potential to get out of date over time

Reply:
Yeah, we should use the `@inheritdoc` natspec tag.